sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java

```diff
@@ -18,38 +17,42 @@
 import java.io.Closeable;
 
 /**
- * CosmosClient with encryption support.
- * We have static method in this class which will takes two inputs
- * {@link CosmosClient} and {@link EncryptionKeyWrapProvider}  and creates cosmosEncryptionClient as shown below.
- * <pre>
- * {@code
- * CosmosEncryptionClient cosmosEncryptionClient =
- * CosmosEncryptionClient.createCosmosEncryptionClient(cosmosClient, encryptionKeyWrapProvider);
- * }
- * </pre>
+ * Provides a client-side logical representation of the Azure Cosmos DB service.
+ * Calls to CosmosClient API's are blocked for completion.
  */
 @ServiceClient(builder = CosmosEncryptionClientBuilder.class)
 public final class CosmosEncryptionClient implements Closeable {
     private final static Logger LOGGER = LoggerFactory.getLogger(CosmosEncryptionAsyncClient.class);
     private final CosmosEncryptionAsyncClient cosmosEncryptionAsyncClient;
-    private EncryptionKeyWrapProvider encryptionKeyWrapProvider;
+    private final KeyEncryptionKeyResolver keyEncryptionKeyResolver;
     private final CosmosAsyncClient cosmosAsyncClient;
     private final CosmosClient cosmosClient;
     private final CosmosClientAccessor cosmosClientAccessor;
+    private final String keyEncryptionKeyResolverName;
 
-    CosmosEncryptionClient(CosmosClient cosmosClient, EncryptionKeyWrapProvider encryptionKeyWrapProvider) {
+    CosmosEncryptionClient(CosmosClient cosmosClient, KeyEncryptionKeyResolver keyEncryptionKeyResolver,
+                           String keyEncryptionKeyResolverName) {
         this.cosmosClientAccessor = CosmosClientHelper.geCosmosClientAccessor();
-        this.encryptionKeyWrapProvider = encryptionKeyWrapProvider;
+        this.keyEncryptionKeyResolver = keyEncryptionKeyResolver;
         this.cosmosClient = cosmosClient;
         this.cosmosAsyncClient = this.cosmosClientAccessor.getCosmosAsyncClient(cosmosClient);
```

Comment:
why we need to hold the reference of both clients? 
any reason any method exposed in CosmosAsyncClient but not in cosmosClient?